presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java

```diff
@@ -883,8 +883,13 @@ private SpecialCallResult tryHandleLike(CallExpression callExpression, List<Obje
                 if (possibleCompiledPattern == null) {
                     return changed(null);
                 }
-                checkState((resolution.isCastFunction(((CallExpression) possibleCompiledPattern).getFunctionHandle())));
-                possibleCompiledPattern = functionInvoker.invoke(((CallExpression) possibleCompiledPattern).getFunctionHandle(), session, nonCompiledPattern);
+
+                checkState(possibleCompiledPattern instanceof CallExpression);
+                // this corresponds to ExpressionInterpreter::getConstantPattern
+                Object[] arguments = hasEscape ? new Object[] {nonCompiledPattern, escape} : new Object[] {nonCompiledPattern};
```

Comment:
I might use `escape != null` rather than `hasEscape`. I actually don't understand why `hasEscape` has to exist. 😂Also I think it's easier to read if you write write an if-else block with two different `functionInvoker.invoke` calls rather than this.

Reply:
Also curious why this part doesn't exist in previous code? -- From the comment it's corresponding to `ExpressionInterpreter::getConstantPattern`? :)

Reply:
@wenleix, I think the original code was always wrong. It didn't respect the original `ExpressionInterpreter::getConstantPattern` semantics. @rongrong, `escape != null` and `hasEscape` should both exist. Because the escape part could be a function that returns a varchar. But the function is on-null-return-null so that the escape part could be null. That will directly nullify the like function as well though it has the escape part.